------------------------------------------------------------
revno: 14121
revision-id: squid3@treenet.co.nz-20161209043304-krtzvsm4a0zbzgi8
parent: squid3@treenet.co.nz-20161209034636-wytrnx7ks2jv0sxt
fixes bug: http://bugs.squid-cache.org/show_bug.cgi?id=4406
author: Michael Buchau <mike@m-buchau.de>
committer: Amos Jeffries <squid3@treenet.co.nz>
branch nick: 3.5
timestamp: Fri 2016-12-09 17:33:04 +1300
message:
  Bug 4406: SIGSEV in TunnelStateData::handleConnectResponse() during reconfigure and restart
------------------------------------------------------------
# Bazaar merge directive format 2 (Bazaar 0.90)
# revision_id: squid3@treenet.co.nz-20161209043304-krtzvsm4a0zbzgi8
# target_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# testament_sha1: ce1153061cb79ac9ede6851f438ec830ed7a3e78
# timestamp: 2016-12-09 04:51:01 +0000
# source_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# base_revision_id: squid3@treenet.co.nz-20161209034636-\
#   wytrnx7ks2jv0sxt
# 
# Begin patch
=== modified file 'src/tunnel.cc'
--- src/tunnel.cc	2016-08-17 13:34:13 +0000
+++ src/tunnel.cc	2016-12-09 04:33:04 +0000
@@ -475,7 +475,8 @@
     *status_ptr = rep.sline.status();
 
     // we need to relay the 401/407 responses when login=PASS(THRU)
-    const char *pwd = server.conn->getPeer()->login;
+    const CachePeer *peer = server.conn->getPeer();
+    const char *pwd = (peer ? peer->login : NULL);
     const bool relay = pwd && (strcmp(pwd, "PASS") == 0 || strcmp(pwd, "PASSTHRU") == 0) &&
                        (*status_ptr == Http::scProxyAuthenticationRequired ||
                         *status_ptr == Http::scUnauthorized);

